x/stake/queryable.go

```diff
@@ -0,0 +1,206 @@
+package stake
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/wire"
+	keep "github.com/cosmos/cosmos-sdk/x/stake/keeper"
+	"github.com/cosmos/cosmos-sdk/x/stake/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+// TODO Redelegations
+func NewQuerier(k keep.Keeper) sdk.Querier {
+	return func(ctx sdk.Context, path []string, req abci.RequestQuery) (res []byte, err sdk.Error) {
+		switch path[0] {
+		case "validators":
+			return queryValidators(ctx, path[1:], k)
+		case "validator":
+			return queryValidator(ctx, path[1:], req, k)
+		case "delegator":
+			return queryDelegator(ctx, path[1:], req, k)
+		case "delegation":
+			return queryDelegation(ctx, path[1:], req, k)
+		case "unbonding-delegation":
+			return queryUnbondingDelegation(ctx, path[1:], req, k)
+		case "delegatorValidators":
+			return queryDelegatorValidators(ctx, path[1:], req, k)
+		case "delegatorValidator":
+			return queryDelegatorValidator(ctx, path[1:], req, k)
+		case "pool":
+			return queryPool(ctx, k)
+		case "parameters":
+			return queryParameters(ctx, k)
+		default:
+			return nil, sdk.ErrUnknownRequest("unknown stake query endpoint")
+		}
+	}
+}
+
+// Params for queries:
+// - 'custom/stake/delegator'
+// - 'custom/stake/delegatorValidators'
+// - 'custom/stake/validator'
+type QueryAddressParams struct {
+	AccountAddr sdk.AccAddress
+}
+
+// Params for queries
+// - 'custom/stake/delegation'
+// - 'custom/stake/unbonding-delegation'
+// - 'custom/stake/delegatorValidator'
+type QueryBondsParams struct {
+	DelegatorAddr sdk.AccAddress
+	ValidatorAddr sdk.AccAddress
+}
+
+func queryValidators(ctx sdk.Context, path []string, k keep.Keeper) (res []byte, err sdk.Error) {
```

Comment:
Great idea. Will open an issue for that since there's a lot to track. I prefer to do it in another separate PR along with the redelegation since wasn't part of the original scope of this PR

Reply:
#2202